Add PostgreSQL & Redis TLS/mTLS support (#3435)
pretix/pretix: doc/admin/config.rst

``user``, ``password``, ``host``, ``port``
    Connection details for the database connection. Empty by default.

``sslmode``, ``sslrootcert``

>     Connection TLS details for the PostgreSQL database connection. Possible values of ``sslmode`` are [disable, allow, prefer, require, verify-ca, verify-full]. Possible value of the ``sslrootcert`` is the accessible path of the ca certificate. Both values are empty by default.

Reviewer: ```suggestion
    Connection TLS details for the PostgreSQL database connection. Possible values of ``sslmode`` are ``disable``, ``allow``, ``prefer``, ``require``, ``verify-ca``, and ``verify-full``. ``sslrootcert`` should be the accessible path of the ca certificate. Both values are empty by default.
```

Author: Done

> ``sslmode``, ``sslrootcert``, ``sslcert``, ``sslkey``

Reviewer: Should we really refer to ``sslmode`` and ``sslrootcert`` here again?

Author: No, you're right. I've simplified it.

>     Connection mTLS details for the PostgreSQL database connection. It's also necessary to specify ``sslmode`` and ``sslrootcert`` parameters inside the mTLS section, please check the correct values from the TLS part. Possible value of the ``sslcert`` is the accessible path of the client certificate.  Possible value of the ``sslkey`` is the accessible path of the client key. All values are empty by default.

Reviewer: ```suggestion
    Connection mTLS details for the PostgreSQL database connection. It's also necessary to specify ``sslmode`` and ``sslrootcert`` parameters, please check the correct values from the TLS part. ``sslcert`` should be the accessible path of the client certificate.  ``sslkey`` should be the accessible path of the client key. All values are empty by default.
```

    If your redis setup doesn't require a password or you already specified it in the location you can omit this option.
    If this is set it will be passed to redis as the connection option PASSWORD.

``ssl_cert_reqs``
    If your redis setup doesn't require TLS or mTLS you can omit this option.

>     If this is set it will be passed to redis as the connection option SSL_CERT_REQS. Possible values are none, optional, required.

Reviewer: For stylistic and spellcheck reasons, we should mark the option name as code probably. Same for the other parameters below.

```suggestion
    If this is set it will be passed to redis as the connection option ``SSL_CERT_REQS``. Possible values are none, optional, required.
```

Author: I've fixed it.

>     If this is set it will be passed to redis as the connection option ``SSL_CERT_REQS``. Possible values are none, optional, required.

Reviewer: ```suggestion
    If this is set it will be passed to redis as the connection option ``SSL_CERT_REQS``. Possible values are ``none``, ``optional``, and ``required``.
```

Author: Done

If they are present they need to be a valid JSON dictionary.
For possible entries in that dictionary see the `Celery documentation`_.

> It is possible the use Redis with TLS/ mTLS for the broker or the backend. If you want to setup it, it is necessary to specify the TLS identifier ``rediss``, the ssl mode ``ssl_cert_reqs`` and optionally specify the CA (TLS) ``ssl_ca_certs``, cert ``ssl_certfile`` and key ``ssl_keyfile`` (mTLS) path as encoded string. the following uri describes the format and possible parameters ``rediss://0.0.0.0:6379/1?ssl_cert_reqs=required&ssl_ca_certs=%2Fetc%2Fpretix%2Fredis-ca.pem&ssl_certfile=%2Fetc%2Fpretix%2Fredis-client-crt.pem&ssl_keyfile=%2Fetc%2Fpretix%2Fredis-client-key.key``

Reviewer: ```suggestion
It is possible the use Redis with TLS/mTLS for the broker or the backend. To do so, it is necessary to specify the TLS identifier ``rediss``, the ssl mode ``ssl_cert_reqs`` and optionally specify the CA (TLS) ``ssl_ca_certs``, cert ``ssl_certfile`` and key ``ssl_keyfile`` (mTLS) path as encoded string. the following uri describes the format and possible parameters ``rediss://0.0.0.0:6379/1?ssl_cert_reqs=required&ssl_ca_certs=%2Fetc%2Fpretix%2Fredis-ca.pem&ssl_certfile=%2Fetc%2Fpretix%2Fredis-client-crt.pem&ssl_keyfile=%2Fetc%2Fpretix%2Fredis-client-key.key``
```

Author: Done
